packages/did-comm/src/didcomm.ts

```diff
@@ -177,14 +178,18 @@ export class DIDComm implements IAgentPlugin {
    * @param transports - A list of {@link IDIDCommTransport} objects. Defaults to
    *   {@link @veramo/did-comm#DIDCommHttpTransport | DIDCommHttpTransport}
    */
-  constructor(transports: IDIDCommTransport[] = [new DIDCommHttpTransport()]) {
+  constructor(
+    transports: IDIDCommTransport[] = [new DIDCommHttpTransport()],
+    isMediateDefaultGrantAll = true,
+  ) {
```

Comment:
Given that `DIDComm` is still marked `@beta` it might be better, given multiple optional parameters with defaults to use an object parameter:
```suggestion
  constructor({
    transports = [new DIDCommHttpTransport()],
    isMediateDefaultGrantAll = true
  }: { transports?: IDIDCommTransport[], isMediateDefaultGrantAll?: boolean }) {
```

Reply:
This affects the following code locations:
```grep
packages/did-comm/src/__tests__/message-handler.test.ts:        new DIDComm([new DIDCommHttpTransport()]),
packages/did-comm/src/__tests__/trust-ping-message-handler.test.ts:        new DIDComm([new DIDCommHttpTransport()]),
packages/did-comm/src/__tests__/routing-message-handler.test.ts:        new DIDComm([new DIDCommHttpTransport()]),
packages/did-comm/src/__tests__/messagepickup-message-handler.test.ts:          new DIDComm([new DIDCommHttpTransport()]),
packages/did-comm/src/__tests__/messagepickup-message-handler.test.ts:          new DIDComm([new DIDCommHttpTransport()]),
__tests__/localAgent.test.ts:      new DIDComm([new DIDCommHttpTransport()]),
__tests__/restAgent.test.ts:      new DIDComm([new DIDCommHttpTransport()]),
```
Where:
```ts
new DIDComm([new DIDCommHttpTransport()])
```
becomes:
```ts
new DIDComm({ transports: [new DIDCommHttpTransport()]})
```

Reply:
It might also be a good idea to add tests for the `isMediateDefaultGrantAll` parameter.

Reply:
I agree this is a cleaner approach. In the `agent.yml` file the syntax would be as follows:

```yaml
- $require: '@veramo/did-comm#DIDComm'
  $args:
    - isDefaultMediateGrantAll: false
      transports:
        $require: '@veramo/did-comm#DIDCommHttpTransport'
```

I have included this and pushed the update. NOTE the implementation of one minor adjustment which is to set the default constructor argument to an empty object allowing for the agent.yml file to exclude these args entirely and still invoke the default configuration as per your implemetation (I checked and without $args the parameter will be undefined).

Updated the invocation of DIDComm with the new argument parameters.